cloudesire-api-client-dto/src/main/java/com/cloudesire/platform/apiclient/dto/model/dto/CloudCredentialDTO.java

```diff
@@ -0,0 +1,78 @@
+package com.cloudesire.platform.apiclient.dto.model.dto;
+
+import io.swagger.annotations.ApiModel;
+import io.swagger.annotations.ApiModelProperty;
+
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+import java.util.Objects;
+
+@ApiModel( description = "Custom cloud credentials" )
+public class CloudCredentialDTO extends NamedEntityDTO
+{
+    @NotNull
+    @Valid
+    private UrlEntityDTO cloudProvider;
+
+    @Valid
+    private UrlEntityDTO owner;
+
+    @ApiModelProperty( value = "Whether the actual credentials have been uploaded", readOnly = true )
+    private Boolean uploaded;
+
+    public CloudCredentialDTO( String name, UrlEntityDTO cloudProvider )
+    {
+        super( name );
+        this.cloudProvider = cloudProvider;
+    }
+
+    public CloudCredentialDTO()
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') "cloudProvider" is marked "javax.validation.constraints.NotNull" but is not initialized in this constructor. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.cloudesire.com/coding_rules#rule_key=squid%3AS2637)